toolz/itertoolz.py

```diff
@@ -496,6 +496,34 @@ def concatv(*seqs):
     return concat(seqs)
 
 
+def duplicate(seq, n=1):
```

Comment:
I find it hard to know the difference between duplicate and repeat naturally. How do you feel about `repeat_each` to connote an elementwise operation. Unfortunatly, `numpy` has prior art for this function but they chose the name `repeat` which classes with the standard lib `itertools`.